### PR TITLE
Add tests for some code generation bugs

### PR DESCRIPTION
**Pull request by alexfikl**
Updated from https://gitlab.tiker.net/inducer/grudge/-/merge_requests/67

### CODE REVIEW

File: test/test_grudge.py [hunk 1]
```diff
@@ -1138,6 +1138,88 @@ def test_empty_boundary(actx_factory):
...
+    div_u = bind(discr, sym_div_u)(actx)
+    error = bind(discr, sym.norm(2, sym.var("x")))(actx, x=div_u - discr.dim)
+    logger.info("error: %.5e", error)
+
+    # }}}
+
+
+@pytest.mark.parametrize("ambient_dim", [
+    2,
+    pytest.param(3, marks=pytest.mark.xfail)
```
Comment (alexfikl) [inline test/test_grudge.py:1171]: This still fails due to some bug in the greedy aggregation.

Comment (inducer) [reply test/test_grudge.py:1171]:
```suggestion
    # FIXME, cf. https://github.com/inducer/grudge/pull/78/
    pytest.param(3, marks=pytest.mark.xfail)
```

**Comment by inducer** [comment]
Thanks for reviving these tests!

**Comment by alexfikl** [comment]
@inducer Did you mean to close this or was it just closed with #76?

**Comment by inducer** [comment]
I didn't mean to close this, sorry. I'm honestly not sure what happened. Maybe it got closed because the base branch for this went away?

**Pull request closed without merging**